noodles-bed/src/de.rs

```diff
@@ -0,0 +1,939 @@
+use std::fmt;
+use std::io::BufRead;
+use std::ops::{AddAssign, MulAssign, Neg};
+
+use noodles_core::Position;
+
+use serde::Deserialize;
+use serde::de::{
+    self, DeserializeSeed, EnumAccess, IntoDeserializer, MapAccess, SeqAccess,
+    VariantAccess, Visitor,
+};
+
+use crate::{Record, reader::Reader, error::{Error, Result}};
+
+impl<'de> Deserialize<'de> for Record<3> {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        enum Field { Chrom, Start, End }
+
+        impl<'de> Deserialize<'de> for Field {
+            fn deserialize<D>(deserializer: D) -> std::result::Result<Field, D::Error>
+            where
+                D: serde::Deserializer<'de>,
+            {
+                struct FieldVisitor;
+
+                impl<'de> Visitor<'de> for FieldVisitor {
```

Comment:
I think it would be clearer to avoid nesting these `impl` blocks.

Reply:
Done, [is it better](https://github.com/umccr/noodles/pull/1/commits/6cc2c6917166f6b5731e410670055f0ff8eb43d3)?

Reply:
Yep, looks good now :+1: